augur/util_support/auspice_config.py

```diff
@@ -0,0 +1,234 @@
+        raise AugurError(f"Config merging for {config_key!r} failed as one (or more) entries were not dictionaries")
+
+    return {**base, config_key: {**base[config_key], **overlay[config_key]}}
+
+def _geo_resolution_id(el: Union[str, dict[str,str]]) -> str:
+    return el['key'] if isinstance(el, dict) else el
+
+def _replace_deprecated(config: dict[str, Any], old_name: str, new_name: Union[None,str], modify: Union[None, Callable]=None):
+    if old_name not in config:
+        return # NO-OP
+    elif new_name is None:
+        deprecated(f"[config file] key {old_name!r} is no longer used and has been dropped from your config")
+        del config[old_name]
+    elif old_name == new_name: # indicates deprecations within an unchanged top-level key
+        assert modify is not None, "[internal error] modify must be a function"
+        config[new_name] = modify(config[old_name])
+    elif new_name in config:
+        deprecated(f"[config file] top level keys {new_name!r} and {old_name!r} were both present, " \
+                   "however the latter is a deprecated version of the former and will be ignored.")
+        del config[old_name]
+    else:
+        deprecated(f"[config file] top level key {old_name!r} is deprecated. Renaming to {new_name!r}" +
+                   (" (with modifications)" if modify else ""))
+        config[new_name] = modify(config[old_name]) if modify else config[old_name]
+        del config[old_name]
+
+def _parse_color_options(options: Any) -> list[dict]:
+    # parse v1-style 'color_options' & convert to v2-style 'colorings'
+    if not isinstance(options, dict):
+        raise AugurError(f"[config file] The deprecated field 'color_options' must be a dictionary")
+    colorings = []
+    for key, info in options.items():
+        # note: if both legentTitle & menuItem are present then we use the latter. See https://github.com/nextstrain/auspice/issues/730
+        if "menuItem" in info:
+            deprecated("[config file] coloring '{}': 'menuItem' has been replaced with 'title'. Using 'menuItem' as 'title'.".format(key))
+            info["title"] = info["menuItem"]
+            del info["menuItem"]
+        if "legendTitle" in info:
+            if "title" in info: # this can only have been set via menuItem ^^^
+                deprecated("[config file] coloring '{}': 'legendTitle' has been replaced with 'title' & is unused since 'menuItem' is present.".format(key))
+            else:
+                deprecated("[config file] coloring '{}': 'legendTitle' has been replaced with 'title'. Using 'legendTitle' as 'title'.".format(key))
+                info["title"] = info["legendTitle"]
+            del info["legendTitle"]
+        if "key" in info:
+            del info["key"]
+        if info.get("type") == "discrete":
+            deprecated("[config file] coloring '{}': type 'discrete' is no longer valid. Please use either 'ordinal', 'categorical' or 'boolean'. "
```

Comment:
```suggestion
            deprecated("[config file] coloring '{}': type 'discrete' is no longer valid. Please use 'ordinal', 'categorical' or 'boolean'. "
```

Reply:
This wasn't introduced by me (it exists in current master) but your suggestion is better & we're now using it